libmscore/element.cpp

```diff
@@ -231,8 +231,15 @@ void Element::deleteLater()
 
 void Element::scanElements(void* data, void (* func)(void*, Element*), bool all)
 {
-    if (all || visible() || score()->showInvisible()) {
-        func(data, this);
+    // if leaf node, apply function on this element (after checking if it is visible)
+    // otherwise, recursively apply scanElements to all children
+    // (Note: This function is overriden in some classes, which may apply `func` even to non-leaf nodes)
```

Comment:
This comment applies to the function as a whole (rather than just part of it) so it would be better to put it before the function as a doc-style comment:

```
//---------------------------------------------------------
//   scanElements
/// If leaf node, apply `func` on this element (after checking if it is visible).
/// Otherwise, recurse over all children (see ScoreElement::scanElements).
/// Note: This function is overridden in some classes to skip certain children,
/// or to apply `func` even to non-leaf nodes.
//---------------------------------------------------------
```

Reply:
done.